uproot4/_util.py

```diff
@@ -635,3 +635,56 @@ def awkward_form_of_iter(awkward1, form):
         )
     else:
         raise RuntimeError("unrecognized form: {0}".format(type(form)))
+
+
+def damerau_levenshtein(a, b, ratio=False):
+    """Modified Damerau-Levenshtein distance. Adds a middling penalty
+    for capitalization.
+
+    Parameters
+    ----------
+    a : str
+    b : str
+    ratio : bool, optional, default False
+        Return ratio (len(a)+len(b) - DLH)/(len(a)+len(b))
+
+    Returns
+    -------
+    float
+        Modified Damerau Levenshtein distance
+    """
```

Comment:
Docstrings are not required for hidden functions like this. I also think it's a little dangerous, since their implementations could change and wrong documentation is worse than no documentation (though people disagree on this point). Anyway, this is written in [NumPy style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) and we use [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

Since it's an internal function, I think a comment would be as good as a docstring, since people should only be using these if they are looking at the code, anyway, so that's up to you. Perhaps the most useful commentary would be the URL where the algorithm is described in detail—that's the sort of thing we'd need if we have to revisit this.